backend/app/testbot/agents/query_understanding.py

```
@@ -0,0 +1,146 @@
+# agents/query_understanding.py
+import logging
+import re
+from typing import Dict, Any, List
```

Ruff (0.8.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)